### DISCUSSION TITLE
building my custom authelia

### DISCUSSION THREAD

**Discussion by ghost** [authelia/authelia#5800] [Q&A] [answered]
Hi,
I downloaded authelia and built it in different ways in ubuntu 22.04. In the last try, I built it using this command:  go build cmd/authelia/main.go. then I ran authelia with this command:  ./authelia --config configuration.yml. when I open 127.0.0.1:9091 in the browser it shows me some json. am I wrong in building authelia? (I tested my configuration by downloading and running the actual authelia execution. it was correct and authelia web UI was ok. )

**Comment by james-d-elliott** [comment]
https://authelia-next.netlify.app/contributing/development/build-and-test/#manually-building

**Reply by ghost** [reply]
I already tried the commands in this link. when I run authelia in this way: ./authelia --config configuration.yml, I get this error: error="failed to load templated assets: open public_html/api/index.html: file does not exist". so authelia can not be run.

**Reply by james-d-elliott** [reply]
What does the output of the `pnpm build` show when you run it?

**Reply by ghost** [reply]
> authelia@4.37.5 build /authelia/web
> vite build

vite v4.4.4 building for production...
✓ 11831 modules transformed.
../internal/server/public_html/index.html                                      1.18 kB │ gzip:   0.57 kB
../internal/server/public_html/static/media/user.84255dd4.svg                  2.10 kB │ gzip:   1.11 kB
../internal/server/public_html/static/media/applestore-badge.4cfabcfd.svg     12.22 kB │ gzip:   4.61 kB
../internal/server/public_html/static/media/googleplay-badge.00ff1bb4.svg     22.80 kB │ gzip:   4.95 kB
../internal/server/public_html/static/css/index.9502ad8b.css                  18.73 kB │ gzip:   2.81 kB
../internal/server/public_html/static/js/index.35854062.js                 1,592.22 kB │ gzip: 641.16 kB

(!) Some chunks are larger than 500 kBs after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/configuration-options/#output-manualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
✓ built in 17.35s

**Reply by james-d-elliott** [reply]
Did you do this step before running the go build last time? If not you must rebuild.

**Reply by ghost** [reply]
yes I did them in the same order in your link.

**Reply by james-d-elliott** [reply]
I'll take a closer look tomorrow, I suspect I missed a step when creating that guide.

**Reply by ghost** [reply]
Hi,
can you please check your guide to find the possible missed step?
thank you

**Reply by james-d-elliott** [reply]
I think it's running (will test later today I think):


```
mv api internal/server/public_html/api
```

**Reply by ghost** [reply]
thank you.
if this command is run how should I have a correct build for authelia?

**Comment by jerryc05** [comment, marked as answer]
> I think it's running (will test later today I think):
> 
> ```
> mv api internal/server/public_html/api
> ```
worked for me! 

`pnpm build` removes `internal/server/public_html/api/`, thus causing the problem. this is how `git status` looks like after `pnpm build`:
```
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	deleted:    internal/server/public_html/api/index.html
	deleted:    internal/server/public_html/api/openapi.yml
	modified:   internal/server/public_html/index.html

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	internal/server/public_html/favicon.ico
	internal/server/public_html/manifest.json
	internal/server/public_html/robots.txt
	internal/server/public_html/static/
```

**Comment by ghost** [comment]
thank you.

I have two questions:
1- Is authelia able to manage (add/edit/delete) users ( usernames and passwords) through it's web UI? currently I know that I should modify a file (for example users_database.yml). Is there a UI manager for this purpose?

2- I need authelia to be integrated with Nginx and be able to manage users' access to different sites. For example, the "s.com" domain is protected by authelia. This domain has four sub-domains named "a","b","c", and "d". Also, two users X and Y are the clients of these sites. After authentication, user X should only be able to access the sub-domains "a" and "b" , and user Y should only be able to access "c" and "d" sites. In other words X can not open "c.s.com" even if it is authenticated (it is not authorized to access "c.s.com"). Is authelia able to handle this case?

**Reply by james-d-elliott** [reply]
1, Not yet.
2. Yes. The ACL's are very granular.

**Comment by ghost** [comment]
thank you.

is it possible to use and configure authelia for different domain names? (for example app.a.com and app.b.com)

**Reply by james-d-elliott** [reply]
Yes

**Reply by ghost** [reply]
Thank you

**Comment by ghost** [comment]
Would you please answer to my last question?
